SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt

```diff
@@ -125,10 +143,21 @@ class ApplicationPresenter : ApplicationContract.Presenter() {
                     parameters["outboundIsArriveBy"] = outboundIsArriveBy
                 }
                 .build()
-            return client.get<DepartureDetails> { url(url) }
-        } catch (e: Exception) {
-            println(e.toString())
+            val departures = client.get<DepartureDetails> { url(url) }
+            if (departures.outboundJourneys.isEmpty()) throw NoDeparturesException()
+
+            return departures
+        } catch (e: ClientRequestException) {
+            val responseText = e.response.readText()
```

Comment:
This could possible be moved inline to line 153, I don't think we gain clarity by making it a variable